plugins/modules/zos_data_set.py

```diff
@@ -1036,16 +1152,53 @@ def data_set_type(contents, dependencies):
     #     return None
     if contents is None:
         return "pds"
+    # raise ValueError (
+    #         vars(dependencies)
+    #     )
+    if contents == "gdg" and dependencies.get("state") == "present" and dependencies.get("limit") is None:
+        raise ValueError(
+            "Limit must be provided when data set type is gd and state=present."
+        )
     types = "|".join(DATA_SET_TYPES)
     if not re.fullmatch(types, contents, re.IGNORECASE):
         raise ValueError(
-            "Value {0} is invalid for type argument. type must be of of the following: {1}.".format(
+            "Value {0} is invalid for type argument. type must be one of the following: {1}.".format(
                 contents, ", ".join(DATA_SET_TYPES)
             )
         )
     return contents
 
 
+def limit_type(contents, dependencies):
```

Comment:
Is this used in the `BetterArgParser`? If so, would it be better to move this function into the arg parser util?

Reply:
I don't think it makes sense to be in BetterArgParser, limit option is only used when creating a gdg without any reference. Like in `zos_copy`, lets say we support copying a gdg base to another gdg base, we can check the limit option for the `src`.
Backup and restore, unarchive should do that automatically as well.